src/cloud/provision/freedom-module.ts

```diff
@@ -0,0 +1,362 @@
+        } else {
+          result.public = val[0];
+          result.private = val[1];
+          this.sendStatus_("SSHKEY_RETRIEVED");
+        }
+        resolve(result);
+      }).catch((err: Error) => {
+        console.error("storage error: " + JSON.stringify(err));
+        reject(err);
+      });
+    });
+  }
+  
+  /**
+   * Make a request to Digital Ocean
+   * @param {String} method - GET/POST/DELETE etc
+   * @param {String} actionPath - e.g. "droplets/"
+   * @param {String} body - if POST, contents to post
+   * @return {Promise.<Object>} - JSON object of response body
+   */
+  private doRequest_ = (method: string, actionPath: string, body?: string) :
+      Promise<Object> => {
+    return new Promise((resolve: Function, reject: Function) => {
+      var url = 'https://api.digitalocean.com/v2/' + actionPath;
+      var xhr = freedom["core.xhr"]();
+      xhr.on("onload", (resolve: Function, reject: Function, xhr: any, e: Error) => {
+        xhr.getResponseText().then((resolve: Function, reject: Function, resp: string) => {
+          try {
+            resolve(JSON.parse(resp));
+          } catch(e) {
+            reject(e);
+          }
+        });
+      });
+      xhr.on("onerror", reject);
+      xhr.on("ontimeout", reject);
+      xhr.open(method, url, true);
+      xhr.setRequestHeader("Authorization", "Bearer " + this.state.oauth.access_token);
+      xhr.setRequestHeader("Content-Type", "application/json");
+      if (body !== null && typeof body !== "undefined") {
+        xhr.send({ string: body });
+      } else {
+        xhr.send(null);
+      }
+    });
+  }
+  
+  /** 
+   * Waits for all in-progress Digital Ocean actions to complete
+   * e.g. after powering on a machine, or creating a VM
+   * @param {Function} resolve - call when done
+   * @param {Function} reject - call on failure
+   */
+  private waitDigitalOceanActions_ = (resolve: Function, reject: Function) : void => {
+    console.log("Polling for Digital Ocean in-progress actions");
+    this.doRequest_("GET", "droplets/" + this.state.cloud.vm.id + "/actions").then((resp: any) => {
+      for (var i = 0; i < resp.actions.length; i++) {
+        if (resp.actions[i].status === "in-progress") {
+          setTimeout(this.waitDigitalOceanActions_, POLL_TIMEOUT);
+          return;
+        }
+      }
+      resolve(resp);
+    }).catch((e: Error) => {
+      console.error("Error waiting for digital ocean actions:" + JSON.stringify(e));
+      reject(e);
+    });
+  }
+  
+  /**
+   * Properly configure Digital Ocean with a single droplet of name:name
+   * Assumes we already have oAuth token and  SSH key in this.state
+   * This method will use this.waitDigitalOceanActions_() to wait until all actions complete
+   * before resolving
+   * @param {String} name of droplet
+   * @return {Promise.<Object>} resolves on success, rejects on failure
+   */
+  private setupDigitalOcean_ = (name: string) : Promise<Object> => {
```

Comment:
Still
```
error TS2453: The type argument for type parameter 'U' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
  Type argument candidate 'Object' is not a valid type argument because it is not a supertype of candidate 'void'.
```
so I determined to do the cast outside.